java/src/test/java/ai/rapids/cudf/BinaryOpTest.java

```diff
@@ -100,6 +99,108 @@ public static ColumnVector forEach(DType retType, ColumnVector lhs, ColumnVector
     }
   }
 
+  private double pmod(double i1, double i2) {
+    double r = i1 % i2;
+    if (r < 0) return (r + i2) % i2;
+    else return r;
+  }
+
+  private long pmod(long i1, long i2) {
+   long r = i1 % i2;
+   if (r < 0) return (r + i2) % i2;
+   else return r;
+  }
+
+  private int pmod(int i1, int i2) {
+    int r = i1 % i2;
+    if (r < 0) return (r + i2) % i2;
+    else return r;
+  }
+
+  @Test
+  public void testPmod() {
```

Comment:
Shorts and Bytes were the ones that required special cases for spark.  It might be good to test those to be sure we don't have to do something different.